### PR TITLE
[WINSPOOL] Implement GetPrinterA based on GetPrinterW

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Implement GetPrinterA

_Add code for GetPrinterA based on GetPrinterW._

JIRA issue: [CORE-16622](https://jira.reactos.org/browse/CORE-16622)

### CODE REVIEW

**Comment by Doug-Lyons** [comment]
@ColinFinck, can you please review?

File: win32ss/printing/base/winspool/printers.c [hunk 1]
```diff
@@ -1031,16 +1031,469 @@ GetDefaultPrinterW(LPWSTR pszBuffer, LPDWORD pcchBuffer)
...
+                StringCchCopyA(ppi7a->pszObjectGUID, cch + 1, pszaObjectGUID);
+
+                HeapFree(hProcessHeap, 0, pszaObjectGUID);
+            }
+        }
+    }       // switch
+
+Cleanup:
+
+    return bReturnValue;
```
Comment (Fraizeraust) [inline win32ss/printing/base/winspool/printers.c]: This doesn't seem like a cleanup body since it just returns the value by the function?

Comment (Doug-Lyons) [reply win32ss/printing/base/winspool/printers.c]: True. This was done to be consistent with the other functions in this program and provides a common return point. It is also the same as the code for EnumPrintersA. Do you have a specific suggestion for a change? Thanks.

**Comment by ColinFinck** [comment]
Thanks for the implementation! Merged your changes.

Allocating a temporary buffer, doing a `WideCharToMultiByte` into that buffer, and then a `StringCchCopyA` seems to be a very recurring pattern for A functions. It should be refactored into an own function.
This should highly reduce the amount of code required and prevent possible copy/paste mistakes.
Would be glad if you could try that for the next A implementation and update the existing ones :)

**Comment by Doug-Lyons** [comment]
Thanks for getting this committed @ColinFinck. Regarding your comment about the recurring pattern, I have already been considering this, but I appreciate your encouragement. I am thinking through how to keep the data type checks and generalize the function as well.

**Pull request merged**